ext/posix/sys/socket.c

```diff
@@ -172,9 +174,13 @@ Psocketpair(lua_State *L)
 }
 
 
-static const char *Safinet_fields[] = { "family", "port", "addr",
+static const char *Safinet4_fields[] = { "family", "port", "addr",
 					/* Also allow getaddrinfo result tables */
 					"socktype", "canonname", "protocol" };
+static const char *Safinet6_fields[] = { "family", "port", "addr",
+					/* Also allow getaddrinfo result tables */
+					"flowinfo", "scope_id",
+					"socktype", "canonname", "protocol" };
```

Comment:
There's a lot of repetition here, and I don't foresee a future where inet4 field names anything but a strict subset of inet6. Why not simplify given that?

```c
static const char *Safinet_fields[] = {
        /* AF_INET6 only */
        "flowinfo", "scope_id",
        /* AF_INET4 and AF_INET6 shared */
        lfamily", "port", "addr",
        /* Also allow getaddrinfo result tables */
        "socktype", "canonname", "protocol" };

#define SAFINET6 0
#define SAFINET4 2
```

and then use `Safinet_fields[SAFINET4]` or similar at use sites?

Reply:
We have to forbid IPv6 fields for functions working with IPv4, so if these fields are present, error should be raised.

Reply:
Using `&Safinet_fields[SAFINET4]` as I propose above would not pass the AF_INET6 fields... but it is only a tiny optimization, which I can do after merging the PR if you'd prefer :)